File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.c
#include <stdbool.h>

int missingMultiple(int* nums, int numsSize, int k) {
    bool seen[10001] = {false};

    for (int i = 0; i < numsSize; i++) {
        seen[nums[i]] = true;
    }

    int multiple = k;

    while (multiple <= 10000 && seen[multiple]) {
        multiple += k;
    }

    return multiple;
}

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var missingMultiple = function(nums, k) {
    const seen = new Set(nums);

    let multiple = k;

    while (seen.has(multiple)) {
        multiple += k;
    }

    return multiple;
};